### PR TITLE
eliminate all VS2017 warnings

### PR DESCRIPTION
**Pull request by zhangyafeikimi**
Hi.

You didn't use type traits with c++ template in a best practice way, so that there are some warnings in VS2017. And some other warnings are about integral type mismatch.

I fixed them all.

### CODE REVIEW

File: include/LightGBM/utils/common.h [hunk 1]
```diff
@@ -271,6 +271,20 @@ inline static std::string ArrayToString(const std::vector<T>& arr, size_t n, cha
   return str_buf.str();
 }
 
+template<typename T, bool is_float>
+struct __StringToTHelper {
```
Comment (guolinke) [inline include/LightGBM/utils/common.h]: StringTo ? or ToString?

Comment (zhangyafeikimi) [inline include/LightGBM/utils/common.h]: __ToStringHelper?

**Comment by zhangyafeikimi** [comment]
@guolinke updated

File: include/LightGBM/utils/common.h (hunk 1 above)
Comment (wxchan) [inline include/LightGBM/utils/common.h]: seem StringTo is right?

Comment (zhangyafeikimi) [inline include/LightGBM/utils/common.h]: StringToTHelper, ToStringHelper are both good names.

Comment (guolinke) [inline include/LightGBM/utils/common.h]: sorry, I mis-understand it. I think StringToT is better.

**Pull request merged**